[Merged by Bors] - Logging, whitespace, comment cleanup

lrettig opened the pull request:
## Motivation
Minor improvements made in the process of testnet debugging. Primarily improves logging and makes some log output cleaner and clearer.

## Changes
- Makes log messages clearer (for testnet debugging)
- Some minor whitespace and comment cleanup
- Some very, very minor logic optimizations

## Test Plan
None, no logic changes

## DevOps Notes
<!-- This section is for DevOps to be aware of changes that might impact releases -->
- [ ] Does this code require configuration changes?
- [ ] Does this code affect public APIs?
- [ ] Does this code assume a new version of external services (PoET, elasticsearch, etc.)
- [x] Does this code make changes to log messages that our monitoring infrastructure may rely on?

lrettig committed:
Improve logging

- Makes log messages clearer (for testnet debugging)
- Some minor whitespace and comment cleanup
- Some very, very minor logic optimizations

lrettig committed:
Make linter happy with strconv

lrettig committed:
Some more minor cleanup

Use defer to close locks, and add another useful debug statement

lrettig committed:
Few more minor log fixes

lrettig committed:
Some more minor fixes

lrettig committed:
Improve tortoise logging

lrettig committed:
Reduce verbosity

lrettig commented:
bors try

lrettig commented:
bors try

lrettig committed:
Clean up msgserver logs

noamnelke commented on cmd/node/node.go:
```diff
@@ -326,11 +326,11 @@ func (app *SpacemeshApp) getAppInfo() string {
 // Cleanup stops all app services
-func (app *SpacemeshApp) Cleanup(cmd *cobra.Command, args []string) {
-	log.Info("App Cleanup starting...")
+func (app *SpacemeshApp) Cleanup(_ *cobra.Command, _ []string) {
```
Removing the names entirely is even cleaner IMO.

```suggestion
func (app *SpacemeshApp) Cleanup(*cobra.Command, []string) {
```

noamnelke commented on common/types/block.go:
```diff
@@ -191,7 +191,7 @@ func (b *Block) Fields() []log.LoggableField {
 		log.Int("view_edges", len(b.ViewEdges)),
 		log.Int("vote_count", len(b.BlockVotes)),
-		b.ATXID,
+		log.String("atxid", b.ATXID.ShortString()),
```
I don't like this change. If we want to print short ATX IDs that's fine, but then we should do it consistently, namely we should change the behaviour of:
```go
// Field returns a log field. Implements the LoggableField interface.
func (t ATXID) Field() log.Field { return t.Hash32().Field("atx_id") }
```

Perhaps all `Hash32`s should be printed shortly in logs, but even if not, we should be consistent with ATX IDs.

noamnelke commented on hare/algorithm.go:
```diff
@@ -394,22 +415,32 @@ func (proc *consensusProcess) handleMessage(m *Msg) {
 	// warn on late pre-round msgs
 	if m.InnerMsg.Type == pre && proc.k != -1 {
-		proc.With().Warning("encountered late PreRound message", log.String("sender_id", m.PubKey.ShortString()), types.LayerID(proc.instanceID))
+		proc.With().Warning("encountered late PreRound message",
```
```suggestion
		proc.With().Warning("encountered late preround message",
```

noamnelke commented on hare/algorithm.go:
```diff
@@ -555,29 +589,30 @@ func (proc *consensusProcess) beginNotifyRound() {
 	cert := proc.commitTracker.BuildCertificate()
 	if cert == nil {
-		proc.Error("Begin notify round: Build certificate returned nil")
+		proc.Error("begin notify round: Build certificate returned nil")
```
```suggestion
		proc.Error("begin notify round: BuildCertificate returned nil")
```

noamnelke reviewed (approved):
❤️❤️❤️

lrettig committed:
Apply suggestions from code review

Co-authored-by: Noam Nelke <noam@spacemesh.io>

lrettig committed:
Apply one more suggestion from review

lrettig committed:
Reduce another genesis error to info

lrettig committed:
Merge branch 'develop' into debugging-logfixes

lrettig commented:
bors merge

lrettig committed:
More small logging improvements

lrettig commented:
bors merge

lrettig committed:
Improve hare logging

lrettig commented:
bors merge

lrettig committed:
Cleanup atxdb, syncer logging

lrettig committed:
Minor logging improvements

lrettig committed:
Update tests to match log output changes

lrettig committed:
Improve block builder logging

lrettig committed:
Clean up block-related logs further

lrettig committed:
Minor block and API-related log cleanup

lrettig committed:
Make linter happy

lrettig committed:
Clean up database logging

lrettig committed:
Update tests to match log output

lrettig commented:
bors merge

lrettig commented:
bors merg

lrettig commented:
bors merge

lrettig commented:
I think system tests are failing due to flakiness (#2222, #2281)

lrettig commented:
bors merge

On Sat, Mar 6, 2021, 23:02 bors[bot] <notifications@github.com> wrote:

> Build failed:
>
>    - ci-stage2
>    <https://github.com/spacemeshos/go-spacemesh/runs/2048979257>
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spacemeshos/go-spacemesh/pull/2296#issuecomment-792196907>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZJWNGPQRN4WTNIAC7EXADTCL3DPANCNFSM4YG2X5PA>
> .
>

lrettig committed:
Log coinbase with ATX

lrettig commented:
bors merg

lrettig commented:
bors merge

lrettig committed:
More minor tweaks to hare and tortoise logging

lrettig committed:
Merge branch 'develop' into debugging-logfixes

lrettig committed:
Make linter happy again

lrettig commented:
bors try

lrettig committed:
Misc logging cleanup

Clean up some incorrect use of LoggableField
Improve logging of incoming message handling

lrettig committed:
Improve logging of hare broker message handling

lrettig committed:
Minor cleanup

lrettig committed:
Fix one more misusage of Field

lrettig commented:
bors merge

lrettig committed:
Make linter happy again

lrettig commented:
bors merge

noamnelke commented on blocks/blockoracle.go:
```diff
@@ -69,16 +71,24 @@ func (bo *Oracle) BlockEligible(layerID types.LayerID) (types.ATXID, []types.Blo
 
 	epochNumber := layerID.GetEpoch()
-	bo.log.Info("asked for eligibility for epoch %d (cached: %d)", epochNumber, bo.proofsEpoch)
+	var cachedEpochDescription log.Field
+	if bo.proofsEpoch == DefaultProofsEpoch {
+		cachedEpochDescription = log.String("cached_epoch_id", "(none)")
+	} else {
+		cachedEpochDescription = log.FieldNamed("cached_epoch_id", bo.proofsEpoch.Field())
+	}
```
@lrettig I know I already approved this and it's already merged, but I just realized that this approach is problematic.

We sometimes put a string (`"(none)"`) and sometimes a numeric value (`bo.proofsEpoch.Field()` uses `log.Uint64()`) in a field named `cached_epoch_id`.

I think that Elasticsearch doesn't like it when a field has more than one data type and it causes indexing issues. Not sure exactly what the implications are.

@narayanprusty is this really an issue? Is there some better practice for cases like this?

lrettig replied:
I agree we need to be careful about this - I didn't realize this was an issue previously, but it caused problems with another field (an error about late message arrival in hare) in the last testnet. It would be good to know the best practice for cases like this.

noamnelke replied:
let's wait for @narayanprusty to chime in

web3rover replied:
If we have different data type for a particular field then ES raises exception and the log is never stored. We have to fix it in code. We can convert uint64 to integer string.

Other solution is to transform the data type using logstash data type transformation plugin. But that isn't a good idea as we have to put install and configure the plugin in CI, spacecraft, local testnet and all other systems that push logs to ELK. In long term it will become difficult to manage this.

noamnelke replied:
@lrettig why is it important to have this `"(none)"` string? I know Kibana let's you filter for explicitly missing fields - can't we just omit the field if it's not cached?

lrettig replied:
Opened #2367, let's follow up there
